FhirToDataLake/SynapseValidate/Synapse_Validator.py

```diff
@@ -0,0 +1,207 @@
+        return sumData
+
+    @staticmethod
+    def parse_continuation_token(response_content):
+        for link_item in response_content["link"]:
+            if link_item["relation"] == "next":
+                parsed_url = urlparse.urlparse(link_item["url"])
+                return urlparse.parse_qs(parsed_url.query)["ct"][0]
+        return None
+
+
+class SchemaManager:
+    def __init__(self, schema_collection_directory):
+        self.schemas = SchemaManager.load_schemas(schema_collection_directory)
+
+    @staticmethod
+    def load_schemas(schema_collection_directory) -> dict:
+        schemas = {}
+        for schema_file in os.listdir(schema_collection_directory):
+            with open(os.path.join(schema_collection_directory, schema_file)) as f:
+                schema = json.load(f)
+                schemas[schema["Type"]] = schema
+        return schemas
+
+    def generate_field_mapping_dicts(self):
+        field_mapping_dicts = {}
+        for resource_type, schema in self.schemas.items():
+            field_mapping_dicts[resource_type] = SchemaManager.generate_external_table_field_mapping_dict(schema)
+        return field_mapping_dicts
+
+    def get_schema(self, resource_type):
+        return self.schemas[resource_type]
+
+    def get_all_resource_types(self):
+        return list(self.schemas.keys())
+
+    @staticmethod
+    def generate_external_table_field_mapping_dict(schema):
+        field_mapping_dict = {}
+        SchemaManager.parse_schema(schema, [], [], field_mapping_dict)
+        return field_mapping_dict
+
+    @staticmethod
+    def parse_schema(node_schema, json_data_path, table_column_path, field_dict_result):
+        if node_schema["IsRepeated"]:
+            field_dict_result[".".join(json_data_path)] = ".".join(table_column_path)
+        elif node_schema["IsLeaf"]:
+            field_dict_result[".".join(json_data_path)] = ".".join(table_column_path)
+        else:
+            for sub_node_name, sub_node_schema in node_schema["SubNodes"].items():
+                table_column_path.append(sub_node_schema["Name"])
+                json_data_path.append(sub_node_schema["Name"])
+
+                SchemaManager.parse_schema(sub_node_schema, json_data_path, table_column_path, field_dict_result)
+
+                table_column_path.pop()
+                json_data_path.pop()
+            return
+
+
+class SQLServerClient:
+    def __init__(self, sql_server_endpoint, database, sql_username, sql_password):
+        self.connectionString = 'DRIVER={ODBC Driver 17 for SQL Server};SERVER=' \
+                    + sql_server_endpoint + ';DATABASE=' + database + ';UID=' + sql_username + ';PWD=' + sql_password
+
+    def get_data(self, resource_type):
+        connection = pyodbc.connect(self.connectionString)
+        sql = f"SELECT * FROM [fhir].[{resource_type}]"
+        data_df = pd.read_sql(sql=sql, con=connection)
+        connection.close()
+        return data_df
+
+
+class DataFactory:
```

Comment:
maybe this is a `FhirSqlDataValidator`? `DataFactory` might be confusing to Azure Data Factory.
We can also move the validation logic from main function to this class.
Maybe we can consider to seperate the DataSource, SchemaManager, Validator to different files?

Reply:
Agree. It looks more like a comparer or validator instead of a data factory.

Reply:
Rename the class to ```DataClient```, I want to use python [ThreadPoolExecutor](https://docs.python.org/3/library/concurrent.futures.html#threadpoolexecutor) here, and it need get returned results from the method, then I want to intuitive compare the data in main method.